framework/include/systems/NonlinearSystemBase.h

```diff
@@ -740,6 +741,8 @@ class NonlinearSystemBase : public SystemBase, public PerfGraphInterface
   void assembleScalingVector();
 #endif
 
+  virtual void postAddResidualObject(ResidualObject &) const {}
```

Comment:
While you _can_ make this method const, from a design perspective it should not be. It modifies objects in the system. If someone has const access to the system, we don't want them to be able to call this!

Please make this non-const.

Reply:
Sounds good. I will let @fande address this. This PR is owned by @fande now ;-)

Reply:
Ha. wrong @fdkong there 😆

Reply:
Changed